### ISSUE TITLE
Unit tests fail to run on grunt server / reload; have unclear messaging when run with grunt test

### ISSUE THREAD

**Issue by facultymatt** [ericclemmons/genesis-skeleton#98]
@ericclemmons I love the new test functionality but I'm having a few issue with getting them to run. 

**1. Unit tests don't run with grunt server or on live reload**
I recall this happening at one point. It seems to have been removed in the current master branch. 

**2. When running grunt test, its not clear what is happening nor does the process resolve itself**
After running `grunt test` the user sees the following in the terminal: 

```
...
Running "karma:unit" (karma) task
INFO [karma]: Karma server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
```

In no way does this indicate what is happening to the end user or what they should do. In fact nothing will happen in the user just sits here. Visiting the page http://localhost:9876/ shows a generic test page. Again its not clear whats happening or what tests are running. Only after clicking "debug" and visiting "http://localhost:9876/debug.html" and opening up the console does the user see the following logs: 

```
SUCCESS homeController should set $scope.client & $scope.server
SUCCESS apiService should return { client: ..., server: ... } 
```

Furthermore, the terminal now logs: 

```
WARN [launcher]: PhantomJS have not captured in 120000 ms, killing.
INFO [launcher]: Trying to start PhantomJS again.
```

In summary,
- Consider adding unit tests back into the grunt server / reload task (or fixing). This might add considerable weight if the user has a lot of tests. Maybe make this optional? 
- Clarify process and messaging when running 'grunt test'. Consider logging the test results and automatically terminating on completion.

**Comment by ericclemmons** [comment]
I'm going to change it it's the following:
- `grunt server` is just for development, no tests.  (I've found things change too much to warrant this)
- `grunt server:unit` will run the server but include the unit test suite.  (For me, this is done to add tests to support/verify normal development behavior)
- `grunt server:e2e` will run the (much more computationally expensive) e2e tests + runner, which would be for resolving cross-platform dependencies.
- `grunt test` would run both of the following
- `grunt test:unit` which is a one-time run of unit tests
- `grunt test:e2e` which is a one-time run of e2e tests

I think this would help differentiate workflows a bit & resolve issues during normal development.

What do you think?

**Comment by ericclemmons** [comment]
I'm about to update `master` with several updates to bower, npm packages, and some of the Gruntfile tasks.

Since Angular 1.1.x came out, and Karma has been updated to 0.9.x, there is significant more setup to have tests working well for both unit + e2e tests.  (YearOfMoo has a good blog post on this, which shows how complex testing is out the gate)

Basically, unit testing will work fine, but I'm going to come back to testing to figure out the easiest way to have both unit, e2e, and RAD together without having to update `index.html` and `karma-*.conf.js` files with every update to your app...

**Comment by ericclemmons** [comment]
`grunt test` is now a one time process.  `grunt server:unit` is effectively the TDD task now.
